### PR TITLE
Fix for Editor Search crash

### LINKED ISSUE #18437: Asset Browser - The Editor crashes when entering anything on search bar
**Describe the bug**
Currently, typing anything on Asset Browser search bar is causing the Editor to crash.

**Assets required**
N/A

**Steps to reproduce**
1. Open up Editor
2. Click on Asset Browser search bar
3. Type anything
4. Note that the Editor crashes

**Expected behavior**
Asset Browser shows the search result instead of crashing

**Actual behavior**
Editor crashes

**Screenshots/Video**
N/A

**Found in Branch**
o3de/main

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
e8570f9d635c0abbf3d733d0524a3c0ae75d726c

**Desktop/Device (please complete the following information):**
 - Device: PC
 - OS: Windows
 - Version: 11 Pro(23H2)
 - CPU: 13th Gen Intel(R) Core(TM) i9-13900K   3.00 GHz
 - GPU: NVidia RTX A4000
 - Memory: 64GB

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by Twolewis**
## What does this PR do?

Fixes https://github.com/o3de/o3de/issues/18437

Was due to a bad/incomplete merge in the asset processor browser table

## How was this PR tested?

Locally, in a debug build running editor.

### CODE REVIEW

File: Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Views/AssetBrowserTableView.cpp [hunk 1]
```diff
@@ -603,7 +603,6 @@ namespace AzToolsFramework
                 clonedFilter->AddFilter(FilterConstType(customTypeFilter));
 
                 clonedFilter->SetFilterPropagation(AssetBrowserEntryFilter::PropagateDirection::Down);
-                m_assetFilterModel->SetFilter(FilterConstType(clonedFilter));
```
Comment (Twolewis) [inline Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Views/AssetBrowserTableView.cpp:606]: This line is what caused the search crash

File: Code/Tools/AssetProcessor/native/ui/BuilderInfoMetricsModel.cpp [hunk 2]
```diff
@@ -23,7 +23,6 @@ namespace AssetProcessor
 
         AZ::s64 dayCount = durationInMs / dayInMs;
         QTime duration = QTime::fromMSecsSinceStartOfDay(durationInMs % dayInMs);
-#if defined(CARBONATED)
```
Comment (Twolewis) [inline Code/Tools/AssetProcessor/native/ui/BuilderInfoMetricsModel.cpp:26]: Removed because a PR of ours got accepted

**Review by Twolewis — COMMENTED** [review]
comments

File: Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Views/AssetBrowserTableView.cpp (hunk 1 above)
Comment (aefimov3) [reply Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Views/AssetBrowserTableView.cpp:606]: I see the same call at the line N 609, I wonder what was the error, double call, too early call?
The removed line looks similar to the "else {}" block where SetFilter follow SetFilterPropgation, if the bug is about double call then maybe we should remove the one at 609. Anyway this is minor, I just curious what the problems was.

Comment (aefimov3) [reply Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Views/AssetBrowserTableView.cpp:606]: For me it works with the first call commented out and with the second call commented out.

Comment (Twolewis) [reply Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Views/AssetBrowserTableView.cpp:606]: I compared against O3DE 2409.1, the second call needs to be there.
see 5295397a09e7f094d38b84792698345bc0259008

**Pull request merged**